comp/core/flare/flare.go

```diff
@@ -132,11 +132,13 @@ func (f *flare) Create(pdata ProfileData, ipcError error) (string, error) {
 	)
 
 	for _, p := range providers {
-		err = p(fb)
-		if err != nil {
-			f.log.Errorf("error calling '%s' for flare creation: %s",
-				runtime.FuncForPC(reflect.ValueOf(p).Pointer()).Name(), // reflect p.Callback function name
-				err)
+		if p != nil {
+			err = p(fb)
+			if err != nil {
+				f.log.Errorf("error calling '%s' for flare creation: %s",
+					runtime.FuncForPC(reflect.ValueOf(p).Pointer()).Name(), // reflect p.Callback function name
+					err)
+			}
```

Comment:
We should filter the list in the constructor instead so the rest of the code can freely use the list without worries.

Also could you add a test for this use case ?